bgpd/bgp_nexthop.c

```diff
@@ -324,6 +325,23 @@ static void bgp_address_del(struct bgp *bgp, struct connected *ifc,
 	}
 }
 
+/* Verify if the bgp router ID is in the address hash table */
+void bgp_validate_set_router_id(struct bgp *bgp, struct in_addr *new_addr)
+{
+	if (!new_addr) {
+		flog_err(EC_BGP_INVALID_RTR_ID,
+			"%s : invalid router ID", __func__);
```

Comment:
Given all callers of this function, `new_addr` can never be NULL so I personally think this check and the `EC_BGP_INVALID_RTR_ID` error are unnecessary. Non-blocking comment.